Loop/Managers/LoopDataManager.swift

```diff
@@ -878,6 +984,89 @@ extension LoopDataManager {
         return prediction
     }
 
+    /// - Throws: LoopError.missingDataError
+    fileprivate func recommendBolus<Sample: GlucoseValue>(forPrediction predictedGlucose: [Sample]) throws -> BolusRecommendation? {
+        guard let glucose = glucoseStore.latestGlucose else {
+            throw LoopError.missingDataError(.glucose)
+        }
+
+        let pumpStatusDate = doseStore.lastAddedPumpData
```

Comment:
[nit] Why separate constance for `doseStore.lastAddedPumpData`, but not `glucose.startDate` (line 997)?